crates/pathfinder/src/storage/state.rs

```diff
@@ -774,13 +774,17 @@ impl StarknetEventsTable {
 
     pub fn event_count(
         tx: &Transaction<'_>,
-        from_block: StarknetBlockNumber,
-        to_block: StarknetBlockNumber,
+        from_block: Option<StarknetBlockNumber>,
+        to_block: Option<StarknetBlockNumber>,
         contract_address: Option<ContractAddress>,
         keys: Vec<EventKey>,
     ) -> anyhow::Result<usize> {
         let mut base_query =
-        r#"SELECT COUNT(1)
+        r#"SELECT COUNT(1),
+           block_number,
+           starknet_transactions.idx as transaction_idx,
+           from_address,
+           starknet_events.keys as keys
```

Comment:
Why are these required? We only seem to use the first column from the result.

Reply:
I think they were required by the joins. Which are themselves not required.